### PR TITLE
feat: akash modules(provider, market, deployment)

### PR DESCRIPTION
**Pull request by huichiaotsou**
## Description

Closes: #XXXX

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: modules/market/module.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/forbole/bdjuno/v3/database"
+	"github.com/forbole/juno/v3/modules"
+)
+
+var (
+	_ modules.Module        = &Module{}
+	_ modules.GenesisModule = &Module{}
+)
+
+// Module represent database/market module
```
Comment (MonikaCat) [inline modules/market/module.go]:
```suggestion
// Module represent x/market module
```

File: cmd/bdjuno/main.go [hunk 2]
```diff
@@ -16,7 +16,9 @@ import (
 	"github.com/forbole/bdjuno/v3/database"
 	"github.com/forbole/bdjuno/v3/modules"
 
-	gaiaapp "github.com/cosmos/gaia/v7/app"
+	// gaiaapp "github.com/cosmos/gaia/v6/app"
```
Comment (MonikaCat) [inline cmd/bdjuno/main.go]:
```suggestion
```

File: modules/deployment/module.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/forbole/bdjuno/v3/database"
+	"github.com/forbole/juno/v3/modules"
+)
+
+var (
+	_ modules.Module        = &Module{}
+	_ modules.GenesisModule = &Module{}
+)
+
+// Module represent database/deployment module
```
Comment (MonikaCat) [inline modules/deployment/module.go]:
```suggestion
// Module represent x/deployment module
```

File: database/provider.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		err := db.saveProviders(paramsNumber, providers, height)
+		if err != nil {
+			return fmt.Errorf("error while storing providers: %s", err)
+		}
+	}
+
+	return nil
+}
+
+// saveProviders allows to store provider inside the database
```
Comment (MonikaCat) [inline database/provider.go]:
```suggestion
// saveProviders allows to store providers inside the database
```

File: hasura/metadata/databases/bdjuno/tables/tables.yaml [hunk 5]
```diff
@@ -22,6 +22,7 @@
 - "!include public_proposal_tally_result.yaml"
 - "!include public_proposal_validator_status_snapshot.yaml"
 - "!include public_proposal_vote.yaml"
+- "!include public_provider.yaml"
```
Comment (MonikaCat) [inline hasura/metadata/databases/bdjuno/tables/tables.yaml:27]: Can you also add metadata for `deployment_params` and `market_params` pls?

**Review by MonikaCat — APPROVED** [review]
tested, LGTM

**Comment by MonikaCat** [comment]
@huichiaotsou Can you update `CHANGELOG.md` as well pls?

**Review by MonikaCat — APPROVED** [review]

**Review by rmontagnin — APPROVED** [review]
utACK

**Pull request merged**